nwb_conversion_tools/utils/genericdatachunkiterator.py

```diff
@@ -0,0 +1,93 @@
+"""Authors: Cody Baker, Saksham Sharda, Oliver Ruebel."""
+from typing import Iterable, Tuple
+import numpy as np
+import psutil
+from abc import abstractmethod
+from hdmf.data_utils import AbstractDataChunkIterator, DataChunk
+from itertools import product
+
+
+class GenericDataChunkIterator(AbstractDataChunkIterator):
+    """DataChunkIterator that lets the user specify chunk shapes."""
+    
+    def _get_chunksize_mb(self, chunk_shape: Iterable, typesize: int) -> float:
+        return np.product(chunk_shape) * typesize / 1e6
+
+    def _set_chunk_shapes(self, chunk_shape: Iterable, typesize: int, buffer_mb: float) -> tuple:
+        n_dim = len(chunk_shape)
+        iter_idx = 0
+        while (self._get_chunksize_mb(chunk_shape, typesize) > buffer_mb) and np.product(chunk_shape) > 2:
+            dim = iter_idx % n_dim
+            chunk_shape[dim] = np.ceil(chunk_shape[dim] / 2.0)
+            iter_idx += 1
+        return tuple(int(x) for x in chunk_shape)
+
+    def __init__(self, chunk_shape: tuple = (), buffer_mb: float = 20.):
```

Comment:
```suggestion
    def __init__(self, chunk_shape: tuple = None, buffer_mb: float = 20.):
```

I generally use `None` for default values. I suppose since a tuple is non-mutable it's not necessarily a problem, but it is a bit of an unusual choice. Is there a particular reason you are using the empty tuple as a default here?

Reply:
Since it's optional you're right, `None` would be better.

Reply:
Fixed!